shoulder/parser/armv8_xml_parser.py

```diff
@@ -129,8 +126,134 @@ def _set_register_access_attributes(self, reg, reg_node):
             else:
                 logger.warn(str(reg.name) + " access_mechanism attribute not found")
 
-        if reg.access_attributes is None:
-            logger.warn(str(reg.name) + " access attributes not found")
+    def _set_register_access_mechanisms(self, reg, reg_node):
+        # Memory mapped access mechanisms
+        reg_address_nodes = reg_node.findall("./reg_address")
+        offsets = set()
+        for node in reg_address_nodes:
+            offset_node = node.find("./reg_offset/hexnumber")
+            offset = int(offset_node.text, 0)
+            offsets.add(offset)
+
+            access_type_nodes = node.findall("./reg_access/reg_access_state/reg_access_type")
+            access_types = set()
+            for access_type_node in access_type_nodes:
+                access_type = access_type_node.text
+                access_types.add(access_type)
+
+            readable_types = set(["RO", "RW", "WI", "IMPDEF", "RO or RW", "RAZ/WI"])
+            writable_types = set(["RW", "WO", "RO or RW", "IMPDEF"])
+
+            if access_types & readable_types:
+                am = model.access_mechanism.ReadMemoryMapped(offset)
+                reg.access_mechanisms.append(am)
+
+            if access_types & writable_types:
+                am = model.access_mechanism.WriteMemoryMapped(offset)
+                reg.access_mechanisms.append(am)
+
+        # Instruction based access mechanisms
+        access_mechanism_nodes = reg_node.findall("./access_mechanisms/access_mechanism")
+        for access_mechanism_node in access_mechanism_nodes:
+            try:
+                accessor = str(access_mechanism_node.attrib["accessor"])
+                operation = accessor.split(' ', 1)[0]
+                operand = accessor.split(' ', 1)[1]
+
+                encoding_nodes = access_mechanism_node.findall("encoding/enc")
+                encoding = {}
+                for enc in encoding_nodes:
+                    name = str(enc.attrib["n"]).lower()
+                    val = int(enc.attrib["v"], 2)
```

Comment:
This causes some issues particularly with the 'n' indexed registers, since "0b[n:2:0]" can't be interpreted as an integer. This shouldn't be an issue for long though, I have a try-except block in my upcoming 'n'-indexed register changes.

Reply:
Agreed, I noticed that this breaks for the 'n' indexed registers as well. I figured this would be the location you would make your changes for parsing the 'n' indexed registers, so I'm guessing you have a better idea than me of how to solve this?